util/dvsim/sim_utils.py

```diff
@@ -99,3 +100,116 @@ def vcs_cov_summary_table(buf):
+            m = re.search(r"total:.*(\d+):(\d+):(\d+)", line)
+            if m:
+                # Convert time format: HR:MIN:SEC to seconds.
+                val = re.findall(r"\d+", m.group(0))
+                job_runtime = (int(val[0]) * 60 + int(val[1])) * 60 + int(val[2])
+                return job_runtime, "s"
+            else:
+                raise SyntaxError("Cannot find job runtime in line: " + line)
+    raise SyntaxError("Cannot find job runtime in this file")
+
+
+def get_simulated_time(log_text: List, tool: str) -> [float, str]:
+    """Capture and return the job simulated_time and its unit.
+
+    The log file is passed as an input to the function, in addition to the tool used. The tool
+    returns a number if the simulated time was extracted successfully. The function will raise
+    appropriate exception if the extraction fails.
+    """
+    if tool == 'xcelium':
+        return xcelium_simulated_time(log_text)
+    elif tool == 'vcs':
+        return vcs_simulated_time(log_text)
+    else:
+        raise NotImplementedError("{tool} is unsupported for run times extraction.")
+
+
+def xcelium_simulated_time(log_text: List) -> [float, str]:
+    """Capture and return the simulated_time from Xcelium simulator.
+
+    Example of Xcelium simulated_time:
+      TODO: use non-UVM dependent printout once the PR is merged.
+      'UVM_INFO @  51524469 ps: (uvm_report_server.svh:847) [UVM/REPORT/SERVER]'
+    """
+    # Reverse the log_text because Xcelium does not print out the total
+    # simulation time in summary, so we parse the log in reversed order to get
+    # the last simulation timestamp.
+    for line in reversed(log_text):
+        # Extract sim runtime from the last timestamp.
+        # TODO: use non-UVM dependent printout once the PR is merged, and
+        # extract the unit as output.
+        if "ps" in line:
+            m = re.search(r"(\d+).*ps", line)
+            if m:
+              return int(m.group(1)), "ps"
+            else:
+                raise SyntaxError("Cannot find the simulation time in line: " + line)
+    raise SyntaxError("Cannot find the simulation time in this log file")
+
+
+def vcs_simulated_time(log_text: List) -> [float, str]:
+    """Capture and return the simulated_time from VCS simulator.
+
+    Example of VCS simulated_time: '$finish at simulation time 522104678 ps'
+    """
+
+    # Reverse the log_text because the simulation time is usually printed at the end of
+    # the file.
+    for line in reversed(log_text):
+        # Extract sim runtime.
+        if "finish at simulation time" in line:
+            # Find pattern: finish at simulation time "XXXX" ps.
+            m = re.search(r"(\d+).+(\b[a-z]+)", line)
+            if m:
+                return int(m.group(1)), m.group(2)
+            else:
+                raise SyntaxError("Cannot find the simulation time in line: " + line)
+    raise SyntaxError("Cannot find the simulation time in the log file")
```

Comment:
Thinking more about this - there are several reasons why this might fail (job killed, etc). Raising a `SyntaxError` exception seems like an overkill, putting extra burden of having to handle the exception. If the job got killed for some reason, we already know something went wrong. Maybe just return `0.0, "ps"` instead?

IMO its great to raise an exception when there is an irrecoverable coding bug - this does not seem like it. 

Ditto for the 4 methods above.

Reply:
okay, that sounds good!